pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -205,9 +208,20 @@ func (r *ReplicaSetReconciler) isStatefulSetReady(mdb mdbv1.MongoDB, expectedUpd
 	return statefulset.IsReady(set, mdb.Spec.Members) && expectedUpdateStrategy == set.Spec.UpdateStrategy.Type, nil
 }
 
+func (r *ReplicaSetReconciler) createService(mdb mdbv1.MongoDB) error {
```

Comment:
I think maybe we're making this a little more complicated that it needs to be, if the goal is to create the service but not worry about errors during creation if the service exists already, we're guarding from those errors by checking for its existence before creating. I think in this case we can replace this function with a call to 

`r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.ServiceName(), Namespace: mdb.Namespace}), svc)`

this will either fail with a conflict error (which we don't consider an error right now), or a different kind of error which we do want to worry about